crates/zksync/compiler/src/zksolc/mod.rs

```diff
@@ -44,33 +56,61 @@ pub struct DualCompiledContracts {
 
 impl DualCompiledContracts {
     /// Creates a collection of `[DualCompiledContract]`s from the provided solc and zksolc output.
-    pub fn new(output: &ProjectCompileOutput, zk_output: &ProjectCompileOutput) -> Self {
+    pub fn new(
+        output: &ProjectCompileOutput,
+        zk_output: &ProjectCompileOutput,
+        artifact_paths: DualCompiledArtifactPaths,
```

Comment:
Yes, because the only thing available to us to index the contracts is the absolute artifact path from foundry's solc artifacts. What we have now, is basically:

```rs
// solc artifact
"/mnt/proj/example-repo/out/Foo.sol/Foo.json"

// zksolc artifact
"/mnt/proj/example-repo/zkout/Foo.sol/Foo.json"
```

so we strip the artifact paths to index `Foo.sol/Foo.json`